components/mpas-framework/src/framework/mpas_pool_routines.F

```diff
@@ -4916,6 +4923,175 @@ subroutine mpas_pool_get_config_logical(inPool, key, value, record)!{{{
    end subroutine mpas_pool_get_config_logical!}}}
 
 
+!-----------------------------------------------------------------------
+!  subroutine mpas_pool_get_config_real_scalar
+!
+!> \brief MPAS Pool Real Config Access Routine
+!> \author Michael Duda, Doug Jacobsen
+!> \date   03/27/2014
```

Comment:
I don't know how picky we want to be about these headers.  But maybe these need correct dates and author(s)?

Reply:
This file was already there before this commit. Do you want me to mark it for when it was created? As best I can tell, it was copied from a file Jacobsen and Duda created in 2013 and Mark Petersen make the copy on May 17, 2021.

Reply:
@brobey, I can definitely understand the confusion.

@mark-petersen and @philipwjones can correct me if I'm wrong but I believe these dates and authors should refer to each subroutine separately, not the whole file.  Further, believe the date is intended to be that of the most recent modification, not the creation of the subroutine.

If my understanding is correct, `\author` should be `Bob Robey` and the `\date` entry should be the date you created each subroutine.

Regarding the May 17, 2021 date on the files, this is merely the date the @mark-petersen moved MPAS-Model over to the E3SM repository, and isn't a significant date in terms of code modifications.

@mark-petersen, we really need an MPAS developer's guide for folks to follow if we want these kind of conventions to be followed as we want.

Reply:
@xylar - I don't know what to do about these. When doing some of the refactoring/rewrites, I started out by adding my name and adding "revised" with a date while preserving the original date. But then I didn't feel like I was actually modifying the algorithms in a lot of cases, so stopped adding/updating. I don't know what we really want to do for this. Acknowledge original author/algorithm, acknowledge all changes? It seems to me we'd like to give credit for the algorithm at a minimum. For minor changes, we have the git record. For significant refactoring that doesn't change the algorithm, I don't know what we want.

Reply:
@philipwjones, I see.  That makes a lot of sense.  I would then suggest that whoever does the refactoring add their name to the list of authors at a minimum.

I think we also need to decide what the date is for.  I don't personally think it's useful for anything at all because git history and git blame can tell you what you need to know better than that (typically inaccurate) date can.  But if we want to keep a "creation date" and a "revised date", that would make sense.

Reply:
How about if I put the following in the heading?
4926 !-----------------------------------------------------------------------
4927 !  subroutine mpas_pool_get_config_real_scalar
4928 !
4929 !> \brief MPAS Pool Scalar Real Config Access Routine
4930 !> \author Bob Robey
4931 !> \creation date   11/16/2021
4932 !> \details
4933 !> This subroutine returns the scalar value associated with a config option with the
4934 !> name key in inPool. This subroutine is based on mpas_pool_get_config_real, but 
4935 !> sets value to be a scalar real instead of a pointer to a real. 
4936 !
4937 !-----------------------------------------------------------------------

Reply:
@xylar @philipwjones I went ahead and fixed the documentation headers for the new routines in mpas_pool_routines.F and mpas_vector_reconstruction.F. I set myself as the author and referenced the original subroutines they were based on.

Reply:
@brobey, thanks very much!  What you've done seems perfect to me.